src/Square.test.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react'
+import renderer from 'react-test-renderer';
+import { Square } from './Square'
+import click from './effects/click'
+import { ClickCell } from './store/actionCreators';
+import { ThemeContext, Themes } from './ThemeContext';
+
+jest.mock('./effects/click')
+
+describe('<Square />', () => {
+    function prepareComponent(inWinDirection = false) {
+        const props = {
+            index: 5,
+            dispatch: jest.fn(),
+            inWinDirection: inWinDirection,
+            value: 'X'
+        };
+
+        const component = renderer.create(
+            <Square
+                {...props}
+            />
+        );
+
+        return {
+            props,
+            component
+        };
+    }
+
+    beforeEach(() => {
+        click.mockClear();
+    });
+
+    it('should render correctly', () => {
+        const { component } = prepareComponent();
+        let tree = component.toJSON();
+        expect(tree).toMatchSnapshot();
+    });
+
+    it('should dispatch a ClickCell action when is clicked', () => {
+        const { props, component } = prepareComponent();
+
+        const tree = component.toJSON();
+        tree.props.onClick();
```

Comment:
```.onClick()``` is just testing a plain JS object (see https://reactjs.org/docs/test-renderer.html#overview). 
It would be more useful to manipulate with DOM since it is how end users act.
[_"The more your tests resemble the way your software is used, the more confidence they can give you."_](https://twitter.com/kentcdodds/status/977018512689455106)

DOM Testing could be done with the help of `react-testing-library`, `Enzyme`, or `react-dom/test-utils`.

Reply:
`react-test-renderer` should be used only to check properties values, to take a one-level snapshot. You should not run callback directly on this objects.

Reply:
Launching click handling by direct call was replaced with DOM event